tests/unit/components/cylc/gscan/gscan.vue.spec.js

```diff
@@ -110,8 +110,24 @@ describe('GScan component', () => {
           [TaskState.FAILED.name]: 1
         }
       }
+
+      // nested mode
       const nodeHierarchy = createWorkflowNode(workflow, true)
-      expect(nodeHierarchy.name).to.equal('a')
+      const ret = [nodeHierarchy.name]
+      const stack = [nodeHierarchy]
+      let snode
+      while (stack.length) {
+        snode = stack.pop()
+        if (snode && snode.children) {
+          for (const child of snode.children) {
+            ret.push(child.name)
+            stack.push(child)
+          }
+        }
+      }
+      expect(ret).to.deep.equal(['a', 'b', 'c'])
```

Comment:
Simpler to do?
```suggestion
      expect(nodeHierarchy).to.deep.nested.include({
        name: 'a',
        'children[0].name': 'b',
        'children[0].children[0].name': 'c'
      })
```

Reply:
It's basically testing for the subset
```js
{
    name: 'a',
    children: [{
        name: 'b',
        children: [{
            name: 'c'
        }]
    }]
}
```
in the object.

Whereas, before #931, it looked like
```js
{
    name: 'a',
    children: [{
        name: 'b',
        children: []
    }, {
        name: 'c',
    }]
}
```
I don't think there should be more than 1 child per level in this function.

I suppose we would also need to check that the `c` node does not have a `children` property